src/test/java/com/myhome/controllers/ControllerTestBase.java

```diff
@@ -0,0 +1,59 @@
+package com.myhome.controllers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.myhome.MyHomeServiceApplication;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT, classes = {
+    MyHomeServiceApplication.class})
+public class ControllerTestBase {
+
+  @Autowired
+  private TestRestTemplate client;
+  @Autowired
+  ObjectMapper objectMapper;
+  @LocalServerPort
+  private int randomPort;
+  private static final String jwtToken = "eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiIyNGI0Mzc5MS02YWY0LTQ5MGYtODYzOS00ZDUyYmJhYjQzNTMiLCJleHAiOjE1OTYyODI4ODl9.ohTsY67FhFp9p79h_rnenHnU6IRxO7-5PojMAhwqNLOM5qlVNSHbpWj-lVTk_fRzk4X-oDcB-o8RP1LZXBI5BQ";
```

Comment:
I do not think it is a good idea to embed a JWT in the test cases. These JWTs would expire in 10 days. So, tests cases will start failing there after.

Can we get it dynamically during login API call and use that for subsequent requests?

Reply:
the idea of JUnits that they are testing small part of the code, I don't want to involve 'login' API in the 'users' API, that might result failures in 'users' tests in case we have issues in other controllers, it is not good practice. is it possible to add a configuration for enabling/disabling jwt validation? that would help us to test all controllers in independent way.

Reply:
You are right, ideally, Unit tests should be independent of each other. But, now the problem is that tests will start failing in a few days. We need to find a solution for this ASAP. (Either mock the JWT parser or introduce dependency between unit tests). Should be fine for now. Let me explore what we can do.